File: kernels/Julia/Compositron/examples/cdbs_fit.jl
using DelimitedFiles

using Compositron.Utils
using Compositron.Core
using Compositron.CDBS


m = Measurement(
    "../../../../testdata/depth-profile_Copper_0000.n42"
)

c = m.cdbs["OAA x OAB"]

correct_ecal!(c, EcalCorrOrder_First)

# S parameter calculation

ls_param = LineshapeParamDefinition(
    "S",
    [DiagonalArea(KeV(2), KeV(1))],
    [DiagonalArea(KeV(10), KeV(1))],
    false
)

calc_lineshape_param!(c, ls_param)

println("S calculation duration")

@time calc_lineshape_param!(c, ls_param)

s = c.lineshape_params["S"]

println("Calculated S parameter: $(s.val) +/- $(s.err)")

# Projection onto diagonal

project_diagonal(c, LinearProjectionBins(KeV(0.1)), KeV(2), KeV(10), false)

println("Projection duration: ")

p = @time project_diagonal(c, LinearProjectionBins(KeV(0.1)), KeV(2), KeV(10), false)

# Write projection to file for inspection

writedlm("projection.csv", p.spectrum, ",")

File: kernels/Python/compositron/examples/cdbs_fit.py
import time

import numpy as np

from compositron.cdbs.cdbspectrum import DiagonalArea, LinearProjectionBins, LineshapeParamDefinition
from compositron.core.measurement import Measurement
from compositron.core.utils import EcalCorrectionOrder, KeV

# Import

m = Measurement(
    "../../../../testdata/depth-profile_Copper_0000.n42"
)

c = m.cdbs["OAA x OAB"]

c.correct_ecal(EcalCorrectionOrder.FIRST)

# S parameter calculation

ls_param = LineshapeParamDefinition(
    name = "S",
    num = [DiagonalArea(KeV(2), KeV(1))],
    denom = [DiagonalArea(KeV(10), KeV(1))],
    in_corrected_peak = False,
)

start = time.time()
s = c.calc_lineshape_param(ls_param)
stop = time.time()

print(f"Calculated S parameter: {s.val} +/- {s.err}");
print(f"Calculation duration: {(stop - start) * 1e6:.1f} µs");

# Projection onto diagonal

start = time.time()
p = c.project_digonal(
    LinearProjectionBins(KeV(0.1)),
    KeV(2.),
    KeV(10.),
    False
)
stop = time.time()

print(f"Projection duration: {(stop - start) * 1e6:.1f} µs");

# Write projection to file for inspection

with open("projection.csv", "w") as f:
    for elem in p.spectrum:
        f.write(f"{elem}\n")
